### PR TITLE
Unique resource lookup

### PR DESCRIPTION
**Pull request by fppt**
- Allow you to get an owner of a specific resource (Only if the resource is unique)
- Allow you to get an instance based on an attached resource (Only if the resource is unique)

### CODE REVIEW

**Comment by fppt** [comment]
please retest this please

**Comment by fppt** [comment]
please retest this please or I will start ignoring more tests

File: mindmaps-graph/src/main/java/io/mindmaps/graph/internal/AbstractMindmapsGraph.java [hunk 1]
```diff
@@ -364,6 +364,11 @@ public Instance getInstance(String id) {
     }
 
     @Override
+    public Instance getInstance(Resource uniqueResource){
+        return uniqueResource.owner();
+    }
```
Comment (aelred) [inline mindmaps-graph/src/main/java/io/mindmaps/graph/internal/AbstractMindmapsGraph.java]: Do we really need this if it only calls a method on the resource passed in?
We should probably be careful when adding more stuff to `MindmapsGraph`.

Comment (fppt) [inline mindmaps-graph/src/main/java/io/mindmaps/graph/internal/AbstractMindmapsGraph.java]: Ya I am not sure. I am in favour of removing it but the requirement asked for a get based on resources.

**Pull request merged**